asyncio_mqtt/client.py

```diff
@@ -1,52 +1,37 @@
 # SPDX-License-Identifier: BSD-3-Clause
+from __future__ import annotations
+
 import asyncio
 import functools
 import logging
 import socket
 import ssl
 import sys
 from contextlib import contextmanager
+from dataclasses import dataclass
 from enum import IntEnum
 from types import TracebackType
 from typing import (
     Any,
-    AsyncContextManager,
     AsyncGenerator,
-    AsyncIterator,
     Awaitable,
     Callable,
     Coroutine,
     Dict,
```

Comment:
I'm not very experienced with typing in Python. Is this on purpose that you're leaving some `typing.Dict`, `typing.Tuple`, and `typing.Union` in the code, instead of changing everything to `dict`, `tuple` and `|`?

Reply:
Yes, the typing used as type annotations can be `dict`, `tuple`, and `|` as they are not executed with `from __future__ import annotations`, but types that are assigned to variables are executed so they must remain as they are.

Reply:
And the non-`typing` versions `dict`, `tuple`, and `|` mustn't be executed because they are not valid python<3.10 code, is that it?

Neat 😎

Reply:
Yes